x-pack/filebeat/input/azureblobstorage/client.go

```diff
@@ -105,13 +106,29 @@ func fetchBlobClientWithSharedKey(url string, credential *azblob.SharedKeyCreden
 func fetchBlobClientWithConnectionString(connectionString string, containerName string, blobName string, log *logp.Logger) (*blob.Client, error) {
 	blobClient, err := blob.NewClientFromConnectionString(connectionString, containerName, blobName, nil)
 	if err != nil {
-		log.Errorf("Error fetching blob client for connectionString : %s, error : %v", connectionString, err)
+		log.Errorf("Error fetching blob client for connectionString: %s, error: %v", stripKey(connectionString), err)
 		return nil, err
 	}
 
 	return blobClient, nil
 }
 
+// stripKey returns the URI part only of a connection string to remove
+// sensitive information. A connection string should look like this:
+//
+//	sb://dummynamespace.servicebus.windows.net/;SharedAccessKeyName=DummyAccessKeyName;SharedAccessKey=5dOntTRytoC24opYThisAsit3is2B+OGY1US/fuL3ly=
```

Comment:
This is also valid with a tail `;EntityPath=<EventHubName>`
https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-get-connection-string

Reply:
So long as it's always the URI first we're good. This comment just reflects what was [here](https://github.com/elastic/beats/blob/14942038f4b8e87ecd8512874844327646437f8f/x-pack/filebeat/input/azureeventhub/v1_input.go#L304-L307).